wasmcloud-test-util/src/lib.rs

```diff
@@ -4,13 +4,72 @@ pub mod provider_test;
 #[cfg(not(target_arch = "wasm32"))]
 pub mod cli;
 
-// re-export testing interface
-pub use wasmcloud_interface_testing as testing;
-
 // re-export regex and nkeys
 pub use nkeys;
 pub use regex;
 
+pub mod testing {
+
+    smithy_bindgen::smithy_bindgen!("testing/testing.smithy", "org.wasmcloud.interface.testing");
+
+    // after sdk is split we won't have to duplicate this code
+    impl Default for TestOptions {
+        fn default() -> TestOptions {
+            TestOptions {
+                patterns: vec![".*".to_string()],
+                options: std::collections::HashMap::default(),
+            }
+        }
+    }
+
+    /// A NamedResult, generated inside a `run_selected!`` or `run_selected_spawn!`` macro,
+    /// contains a tuple of a test case name and its result. The implementation of From here
+    /// makes it easy to use the `into()` function to turn the NamedResult into a TestResult.
+    pub type NamedResult<'name, T> = (&'name str, RpcResult<T>);
+
+    // convert empty RpcResult into a testResult
+    impl<'name, T: Serialize> From<NamedResult<'name, T>> for TestResult {
+        fn from(name_res: NamedResult<'name, T>) -> TestResult {
+            let test_case_name = name_res.0.to_string();
+            let test_case_result = name_res.1;
+            match test_case_result {
+                Ok(res) => {
+                    // test passed. Serialize the data to json
+                    let data = match serde_json::to_vec(&res) {
+                        Ok(v) => serde_json::to_vec(&serde_json::json!({ "data": v }))
+                            .unwrap_or_default(),
+                        // if serialization of data fails, it doesn't change
+                        // the test result, but serialization errors should be logged.
+                        // Logging requires us to have logging set up, but since we might be running as an actor,
+                        // and we can't force the user to add a logging dependency and set up logging,
+                        // so there isn't much we can do here. Not even println!().
+                        Err(_) => b"".to_vec(),
+                    };
+                    TestResult {
+                        name: test_case_name,
+                        passed: true,
+                        snap_data: Some(data),
+                    }
+                }
+                Err(e) => {
+                    // test failed: generate an error message
+                    let data = serde_json::to_vec(&serde_json::json!(
+                        {
+                           "error": e.to_string(),
+                        }
+                    ))
+                    .ok();
+                    TestResult {
+                        name: test_case_name,
```

Comment:
```suggestion
        fn from((name, res): NamedResult<'name, T>) -> TestResult {
            let name = name.into();
            match res {
                Ok(res) => {
                    // test passed. Serialize the data to json
                    let data = match serde_json::to_vec(&res) {
                        Ok(v) => serde_json::to_vec(&serde_json::json!({ "data": v }))
                            .unwrap_or_default(),
                        // if serialization of data fails, it doesn't change
                        // the test result, but serialization errors should be logged.
                        // Logging requires us to have logging set up, but since we might be running as an actor,
                        // and we can't force the user to add a logging dependency and set up logging,
                        // so there isn't much we can do here. Not even println!().
                        Err(_) => b"".to_vec(),
                    };
                    TestResult {
                        name,
                        passed: true,
                        snap_data: Some(data),
                    }
                }
                Err(e) => {
                    // test failed: generate an error message
                    let data = serde_json::to_vec(&serde_json::json!(
                        {
                           "error": e.to_string(),
                        }
                    ))
                    .ok();
                    TestResult {
                        name,
```